Kernel/PartialOrdering.cpp

```diff
@@ -0,0 +1,648 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+/**
+ * @file PartialOrdering.cpp
+ * Implements class PartialOrdering.
```

Comment:
Could I have a couple lines explaining the approach taken here? I liked the one for `OrderingComparator`.